nixos/modules/services/hardware/auto-cpufreq.nix

```diff
@@ -0,0 +1,34 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.auto-cpufreq;
+in {
+  options = {
+    services.auto-cpufreq = {
+      enable = mkEnableOption "auto-cpufreq daemon";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.auto-cpufreq ];
+
+    systemd.services.auto-cpufreq = {
+      description = "auto-cpufreq - Automatic CPU speed & power optimizer for Linux";
+
+      # XXX: This is in the upstream .service and I don't know why.
+      after = [ "network.target" "network-online.target" ];
+
+      # XXX: The daemon calls a bash script and pipes clear to the log file.
+      path = with pkgs; [ ncurses bash ];
+
+      serviceConfig = {
+        Type = "simple";
+        User = "root";
+        ExecStart = "${pkgs.auto-cpufreq}/bin/auto-cpufreq --daemon";
+        StandardOutput = "append:/var/log/auto-cpufreq.log";
+      };
+      unitConfig.ConditionPathExists = "/var/log/auto-cpufreq.log";
```

Comment:
Could we easily log to journal?

Reply:
This was just copied from the [upstream service file](https://github.com/AdnanHodzic/auto-cpufreq/blob/master/scripts/auto-cpufreq.service). it pipes `clear` to the file, so when you use `cat` it clears the previous logs. It spams the logs with a ton of info like current distro, kernel, and cpu state. I may just get rid of the logs, since they keep filling up with useless info.

Reply:
If you're just copy+pasting the upstream unit you should just directly include it via `systemd. packages`. We'll stay in sync with upstream that way.

Reply:
The problem is that the package doesn't currently include the service file. Should I include it in this PR as a separate commit?

Reply:
Yes. That sounds ideal :+1: